src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -2851,6 +2851,138 @@ static void Main()
             comp.VerifyEmitDiagnostics();
         }
 
+        [Fact]
+        public void TypeInference_Spread_09()
```

Comment:
Are these new tests relevant to the implementation changes in this PR? It looks like the implementation change only affects error reporting.

Reply:
No, these tests are not related to the changes in the PR.